packages/toolpad-app/src/toolpadDataSources/rest/server.ts

```diff
@@ -138,6 +139,21 @@ async function execBase(
     resolveBindableEntries(fetchQuery.headers || [], params),
   ]);
 
+  if (process.env.TOOLPAD_DEMO) {
+    const demoUrls = DEMO_BASE_URLS.map((baseUrl) => baseUrl.url);
+
+    const hasNonDemoConnectionParams =
+      !demoUrls.includes(connection?.baseUrl || '') ||
+      (connection?.headers && connection?.headers.length > 0) ||
```

Comment:
Would `connection?.headers?.length > 0` work the same way? 🤔

Reply:
It's not doing `undefined > 0` it's "short circuiting" and simply returning `undefined` if `headers` are not defined, so should be doing pretty much the same thing just gonna be shorted: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

IMO it's weird to use that on `connection` but not use that on `headers` since that's why the operator exists

Reply:
ah my bad, i think `connection?.headers.length > 0` is also not good...
thanks for pointing this out, i'll improve it.

Reply:
i've changed it, i think it should be better now?

Reply:
I'm not sure I understand, what exactly has been changed? addition of `!!` in front? That shouldn't really change anything since `undefined` is falsy value anyways and doing `!!` on undefined would be same in all cases. I suggest reading on optional chaining and how it works it might clear up the confusion here or you could see what is the result of compiled `connection?.headers?.length > 0` [here in babel repl](https://babeljs.io/repl/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYewdmCmwC4JbgPwDoAWkCGATSAnAzigDaRgDmMqABAHxUAMQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.19.2&externalPlugins=&assumptions=%7B%7D)

Reply:
i know about optional chaining, i just added `!!` so that the variable is a `boolean` instead of `undefined` just for it to be cleaner, but that change was not to address your comment.

what changed is that i got rid of the `?` in `connection?.headers.length > 0` because it results in `undefined > 0` which does result to `false` but isn't good logic, and the optional chaining wasn't needed there.

Reply:
Not exactly sure what is meant by "isn't good logic" - do you see where it could break or behave in a way that is not expected if you used optional chaining? Because if left side evaluates to `undefined` the entire condition will always be false. The only case when it would be true is if everything was defined and length was more than 0 - exactly what you want.
Anyways if you prefer to keep it as is I'm fine, I just wanted to point out that same can be achieved with `connection?.headers?.length > 0` which is shorter and (I personally find it) easier to understand way. Unless I'm totally wrong here and it can actually break under some circumstances that I'm not familiar with?

Reply:
yeah i just didn't want to use "javascript magic" and have more explicit types, but doesn't mean anything would break - just personal preference. 
doing it like you say is less verbose for sure though - probably a bit easier to read.
i don't mind changing it to your suggestion, it's just a style difference really.

Reply:
actually just tired your suggestion and Typescript complains about it so let's keep it as is if it's ok!

Reply:
Sure, let's keep it as is 👍 . Would you mind sharing what sort of error/warning TS threw? Wanna know what to look for in the future 🤔 
I'm trying to guess that it complains comparing `undefined to number` 🤔  In that case something like that could be used:
```
(connection?.headers?.length || 0) > 0
```

Reply:
<img width="633" alt="Screen Shot 2022-09-20 at 18 59 35" src="https://user-images.githubusercontent.com/10789765/191330709-df184001-c5ae-464b-b288-4eca63f9d2ff.png">

Reply:
Thanks! My bad, didn't compile TS in my head 😅 anyways, it was an interesting find (regarding TS) - seems that there is even an open issue discussing this on TS repo: https://github.com/microsoft/TypeScript/issues/45543 maybe in the future this will change who knows :)